### ISSUE TITLE
cliArgs can't be accessed in constructor

### ISSUE THREAD

**Issue by saikrishna321** [appium/appium#17327]
## The problem

Unable to access this.cliArgs from plugin constructor. It returns empty object. 

## Environment

* Appium version (or git revision) that exhibits the issue: latest beta.42

## Code To Reproduce Issue [ Good To Have ]

```
constructor(name) {
    super();
    this.name = name;
    console.log(`hello: ${JSON.stringify(this.cliArgs, null, 2)}`);
  }
```

**Comment by boneskull** [comment]
@saikrishna321 I'm going to need more information as the unit and E2E tests pass for this behavior.  I will need:

- A complete plugin exhibiting the behavior, including `package.json`
- Driver(s) used
- The CLI arguments or configuration passed into Appium

**Comment by saikrishna321** [comment]
@boneskull Repo link - https://github.com/AppiumTestDistribution/appium-wait-plugin/tree/cliArgs

```
npm run build && npm install

./node_modules/.bin/appium server -ka 800 --use-plugins=element-wait --config /Users/saikrisv/Documents/git/appium-wait-plugin/server-config.json -pa /wd/hub --plugin-element-wait-interval-between-attempts=2000
```

server-config.json - https://github.com/AppiumTestDistribution/appium-wait-plugin/blob/cliArgs/server-config.json

Let me know if you need more information.

**Comment by boneskull** [comment]
@saikrishna321 Sorry; what request did you send to trigger the plugin?

I see a couple problems here otherwise:

1. Your plugin should have a peer dependency on Appium `"appium": "^2.0.0-beta.42"`
2. Your plugin should _not_ depend on `@appium/support` nor `@appium/base-plugin`
3. Replace `require('@appium/support')` and `require('@appium/base-plugin)` with `require('appium/suppot')` and `require('appium/plugin')`, respectively

**Comment by saikrishna321** [comment]
@boneskull Thanks for looking at the repo and giving your recommendations. 

I have addressed Point 1 and 2 -https://github.com/AppiumTestDistribution/appium-wait-plugin/blob/cliArgs/package.json#L65

Point 3: It failed when I made changes as recommended. 
```
[Appium] Attempting to load plugin element-wait...
[debug] [Appium] Requiring plugin at /Users/saikrisv/Documents/git/appium-wait-plugin/node_modules/appium-wait-plugin
[Appium] Could not load plugin 'element-wait', so it will not be available. Error in loading the plugin was: Cannot find module 'appium/base-plugin'
```

Keeping it `import BasePlugin from '@appium/base-plugin';` works 

`what request did you send to trigger the plugin?` I have a e2e test written with WDIO. https://github.com/AppiumTestDistribution/appium-wait-plugin/blob/cliArgs/__tests__/e2e.spec.js

`npm run test-e2e` you need a device/emulator connected. Let me know if there is a better way to test it e2e.

**Comment by boneskull** [comment]
Hmm, that looks like a separate issue and I'll look into it.  As a workaround for testing until it's fixed:

1. ensure you're using npm v7 or newer
2. make the changes above to your `package.json` (peer dep of `appium`, remove `@appium/base-plugin`,` @appium/support`)
3. run `npm ci` in your working copy of `appium-wait-plugin` to reinstall everything

Then:

```bash
# declare a temporary dir for APPIUM_HOME
export APPIUM_HOME=/tmp/test-appium-wait-plugin
# install plugin from working copy into APPIUM_HOME
./node_modules/.bin/appium plugin install --source=local .
# install driver
./node_modules/.bin/appium driver install uiautomator2
# start server
./node_modules/.bin/appium server -ka 800 --use-plugins=element-wait --config ./server-config.json -pa /wd/hub --plugin-element-wait-interval-between-attempts=2000
```

**Comment by saikrishna321** [comment]
```
ensure you're using npm v7 or newer
make the changes above to your package.json (peer dep of appium, remove @appium/base-plugin, @appium/support)
```

I have made these changes. 

Let me know once the fix is done to get the cliArgs from the constructor. Will re-test.

**Comment by boneskull** [comment]
See https://github.com/AppiumTestDistribution/appium-wait-plugin/pull/31 which tests against `@appium/fake-driver`.  That PR is not intended to be merged, btw. 🙂
